File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/UserCouponJpaRepository.java
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository;

import com.eager.questioncloud.core.domain.coupon.infrastructure.entity.UserCouponEntity;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface UserCouponJpaRepository extends JpaRepository<UserCouponEntity, Long> {
    Boolean existsByUserIdAndCouponId(Long userId, Long couponId);

    Optional<UserCouponEntity> findByIdAndUserIdAndIsUsedFalse(Long id, Long userId);
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/coupon/infrastructure/repository/UserCouponJpaRepository.kt
package com.eager.questioncloud.core.domain.coupon.infrastructure.repository

import com.eager.questioncloud.core.domain.coupon.infrastructure.entity.UserCouponEntity
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.*

@Repository
interface UserCouponJpaRepository : JpaRepository<UserCouponEntity, Long> {
    fun existsByUserIdAndCouponId(userId: Long, couponId: Long): Boolean

    fun findByIdAndUserIdAndIsUsedFalse(id: Long, userId: Long): Optional<UserCouponEntity>
}
